### PR TITLE
WIP: Stop-Process: add parameter 'IncludeChildProcess'

### PR DESCRIPTION
**Pull request by nickkimbrough**
# PR Summary
In issue #15075, it was requested that `Stop-Process` have functionality to stop child processes. This PR adds the parameter `IncludeChildProcess` in order to accomplish that.

## PR Context

This PR will address #15075 by adding the discussed parameter.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: MicrosoftDocs/PowerShell-Docs#8388
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe "Stop-Process" -Tags "CI" {
+    # Parameter validation
+    $command = Get-Command -Name Stop-Process
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1:6]: Please put in BeforeAll block.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    It "Should have an Id parameter" {
+        $command | Should -HaveParameter "Id" -Type [int[]] -Mandatory $false -DefaultValue $null
+    }
+
+    It "Should have an InputObject parameter" {
+        $command | Should -HaveParameter "InputObject" -Type [System.Diagnostics.Process[]] -Mandatory $false -DefaultValue $null
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1:13]: We never do so formal tests.
Please add _functional_ tests.

Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1:13]: There should be a test where you create a process and a couple of child processes and verify that if the switch isn't used, the child processes don't get killed, but if the switch is used, the entire tree is killed.  Also, should have a test where a child process is not killed to verify the error handling.  Not sure if you can reliability create such a situation so you may need to have a test hook.  (search for test hooks in our current tests to see how they are used and how they are implemented)

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1:13]: I guess the child process could be pwsh with Sleep 600.

Comment (nickkimbrough) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1:13]: When writing the WIP PR I had similar thoughts, and pwsh also came to mind. I am however not quite as familiar with PowerShell on Mac and Linux. Would pwsh be the same on every environment?

Comment (nickkimbrough) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1:13]: It looks like it is, I will try this.

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1:13]: Windows and Unix-s differ in how they manage processes. You will most likely see differences in behavior and have differences in tests. For this reason, Steve mentioned hooks as a clue.

**Pull request closed without merging**